support-lib/jni/djinni_support.cpp

```diff
@@ -239,8 +239,13 @@ jint JniEnum::ordinal(JNIEnv * env, jobject obj) const {
 
 LocalRef<jobject> JniEnum::create(JNIEnv * env, jint value) const {
     LocalRef<jobject> values(env, env->CallStaticObjectMethod(m_clazz.get(), m_staticmethValues));
+    jniExceptionCheck(env);
     DJINNI_ASSERT(values, env);
-    return LocalRef<jobject>(env, env->GetObjectArrayElement(static_cast<jobjectArray>(values.get()), value));
+    LocalRef<jobject> result(env,
+                             env->GetObjectArrayElement(static_cast<jobjectArray>(values.get()),
+                                                        value));
+    jniExceptionCheck(env);
+    return std::move(result);
```

Comment:
Variables *declared* as `const` must never be accessed as non-`const`. The reason is that whenever you get a `const&` function argument that variable might be placed in readonly memory. You simply don't know, so mutating it might destroy the world. In case 1 the callee's version is *declared*  as `const`, and since with RVO the variable in the caller is the *same variable* it would mutate a variable originally declared `const`. That violates the rules.

> Or put a different way, if mutating a const value could be unsafe, why couldn't the compiler effectively create a non-const xtmp for RVO, and treat x like a const reference to it.
That is exactly what happens in case 2. But you cannot swap it: it's illegal to create a `const` xtmp and access it as non-`const&` x as would happen in case 1.